client/client.go

```diff
@@ -163,3 +166,46 @@ func (client Mikrotik) getMikrotikClient() (c *routeros.Client, err error) {
 
 	return
 }
+
+func boolToMikrotikBool(b bool) string {
+	if b {
+		return "yes"
+	} else {
+		return "no"
+	}
+}
+
+func Marshal(s interface{}) string {
+	rv := reflect.ValueOf(s)
+	elem := rv.Elem()
+
+	if rv.Kind() != reflect.Ptr {
+		panic("Command attribute construction cannot work without a pointer")
```

Comment:
I don't think this is true and I think it's due to your implementation (using `rv.Elem()` rather than `rv`). `Unmarshal`ing requires a pointer because editing a non-pointer variable in a function means the edits will be reverted once the function returns (since a function call will get its own copy of the variable). Let me know if that doesn't make sense (and [this blog post](https://goinbigdata.com/golang-pass-by-pointer-vs-pass-by-value/) explains the difference between the two).

This isn't the case for Marshal and it would be good if we could get it to accept both types of inputs (pointer and non pointer). The trick will be using `rv.Elem()` when `s` is a pointer and using `rv` when `s` is not a pointer.

Reply:
Ah, yes I understand. I've addressed this and adapted a unit test to reflect the functionality.